pysat/instruments/sw_kp.py

```diff
@@ -284,26 +284,29 @@ def download(date_array, tag, sat_id, data_path, user=None, password=None):
         ftp = FTP('ftp.gfz-potsdam.de')   # connect to host, default port
         ftp.login()               # user anonymous, passwd anonymous@
         ftp.cwd('/pub/home/obs/kp-ap/tab')
+        dnames = list()
 
         for date in date_array:
             fname = 'kp{year:02d}{month:02d}.tab'
             fname = fname.format(year=(date.year - date.year//100*100),
                                  month=date.month)
             local_fname = fname
             saved_fname = os.path.join(data_path, local_fname)
-            try:
-                print('Downloading file for '+date.strftime('%x'))
-                sys.stdout.flush()
-                ftp.retrbinary('RETR '+fname, open(saved_fname, 'wb').write)
-            except ftplib.error_perm as exception:
-
-                if str(exception.args[0]).split(" ", 1)[0] != '550':
-                    raise
-                else:
-                    # file isn't actually there, just let people know
-                    # then continue on
-                    os.remove(saved_fname)
-                    print('File not available for '+date.strftime('%x'))
+            if not fname in dnames:
+                try:
+                    print('Downloading file for '+date.strftime('%b %Y'))
+                    sys.stdout.flush()
+                    ftp.retrbinary('RETR '+fname, open(saved_fname, 'wb').write)
+                    dnames.append(fname)
+                except ftplib.error_perm as exception:
+
+                    if str(exception.args[0]).split(" ", 1)[0] != '550':
+                        raise
```

Comment:
Should we specify what we're raising here?

Reply:
The bare raise preserves the stack trace back to the issue with ftp.

https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python

Reply:
Good to know.  Perhaps adding that as a comment for future developers would be useful and help people adding new instruments.

Reply:
Comment has been added